pydra/engine/core.py

```diff
@@ -548,13 +572,14 @@ def _run(self, rerun=False, environment=None, **kwargs):
                 save(output_dir, result=result, task=self)
                 # removing the additional file with the checksum
                 (self.cache_dir / f"{self.uid}_info.json").unlink()
-                # # function etc. shouldn't change anyway, so removing
-                orig_inputs = {
-                    k: v for k, v in orig_inputs.items() if not k.startswith("_")
-                }
-                self.inputs = attr.evolve(self.inputs, **orig_inputs)
+                # Restore original values to inputs
+                for field_name, field_value in orig_inputs.items():
+                    setattr(self.inputs, field_name, field_value)
```

Comment:
why not call `attr.evolve` directly here if the startswith is not relevant in construction?

Reply:
No reason, I just didn't think about it after I had changed a few things back and forth. Updated it now

Reply:
I remembered/realised why I went with setattr instead of evolve, because the `_hashes` attribute I'm using to store the original hashes before the run gets wiped out (did a hard reset to previous commit to revert)